common-cli/src/main/java/co/cask/common/cli/completers/StringsCompleter.java

```diff
@@ -36,13 +39,14 @@
   @Override
   protected Collection<String> getCandidates(String buffer) {
     ImmutableList.Builder<String> candidates = ImmutableList.builder();
-
+    ArrayList<String> mutableCandidates = new ArrayList<String>();
     for (String candidate : getStrings()) {
       if (candidate.startsWith(buffer)) {
-        candidates.add(candidate);
+        mutableCandidates.add(candidate);
       }
     }
-
+    Collections.sort(mutableCandidates, Ordering.usingToString());
+    candidates.addAll(mutableCandidates);
```

Comment:
Copying to an immutable list is unnecessary. Just have

`return Collections.unmodifiableList(mutableCandidates);`